index.js

```diff
@@ -137,6 +138,34 @@ class Realm {
       await model.sync();
     }
   }
+
+  query(query, values, opts = {}) {
+    return this.driver.query(query, values, opts);
+  }
+
+  async transaction(callback) {
+    if (callback.constructor.name !== 'GeneratorFunction') {
+      throw new Error('unexpected transaction function, should be GeneratorFunction.');
+    }
```

Comment:
```suggestion
    if (callback.constructor.name === 'AsyncFunction') {
      const connection = await this.driver.getConnection();
      await callback({ connection });
      return;
    }
```

或许可以补一个类似这样的判断来支持 async function（这个示例是简化的，还漏掉了 begin / commit / rollback 判断

另外这里的 transaction 实现和 Bone 里面的最好合并一下，避免维护两个版本